### PR TITLE
use the old reddit mobile site on Android and iOS

### PR DESCRIPTION
**Pull request by oconnor663**
r? @cjb 

It looks like we construct the URL here on the client side, so making the change on the server would require much bigger changes. Would this work for you?

### CODE REVIEW

**Comment by oconnor663** [comment]
Might want to get your sanity check as well, @maxtaco.

File: go/externals/proof_support_reddit.go [hunk 1]
```diff
@@ -99,9 +100,18 @@ func (t RedditServiceType) FormatProofText(ppr *libkb.PostProofRes) (res string,
 
 	q := urlReencode(libkb.HTTPArgs{"title": libkb.S{Val: title}, "text": libkb.S{Val: ppr.Text}}.EncodeToString())
 
+	// The new reddit mobile site doesn't respect the post-pre-populate query
+	// parameters. Use the old mobile site until they fix this.
+	var host string
+	if libkb.IsIOS || runtime.GOOS == "android" {
```
Comment (maxtaco) [inline go/externals/proof_support_reddit.go]: @patrickxb isn't there a better way to tell if we're on mobile?

Comment (patrickxb) [inline go/externals/proof_support_reddit.go]: ```r.G().Env.GetAppType() == libkb.MobileAppType```

Comment (maxtaco) [inline go/externals/proof_support_reddit.go]: can pass this through the ProofContext.

Comment (oconnor663) [inline go/externals/proof_support_reddit.go]: I saw the `DesktopAppType` constant, but it looks like that's only ever returned from `libcmdline`? I wasn't able to tell quite what was going on there. Let me know if I should try to use those constants.

**Comment by cjb** [comment]
The URL rewrite worked, though Reddit seems to be trying to give me a captcha while not actually rendering it on that page, so this solution seems pretty fragile.  Still, okay to merge in that it seems better than what we have now..

**Comment by oconnor663** [comment]
Rewrote with a different approach, passing `GetAppType` through the `ProofContext` like you guys suggested. Could you take another look, @maxtaco and @patrickxb?

**Comment by oconnor663** [comment]
@maxtaco could you eyeball the context passing for me, and make sure it looks right?

**Pull request merged**